zebra-chain/src/sapling/output.rs

```diff
@@ -27,6 +27,30 @@ pub struct Output {
     pub zkproof: Groth16Proof,
 }
 
+impl Output {
+    /// Encodes the primary inputs for the proof statement as 5 Bls12_381 base
+    /// field elements, to match bellman::groth16::verify_proof.
+    ///
+    /// NB: jubjub::Fq is a type alias for bls12_381::Scalar.
+    ///
+    /// https://zips.z.cash/protocol/protocol.pdf#cctsaplingoutput
+    pub fn primary_inputs(&self) -> Vec<jubjub::Fq> {
+        let mut inputs = vec![];
+
+        let cv_affine = jubjub::AffinePoint::from_bytes(self.cv.into()).unwrap();
```

Comment:
can we replace these `unwrap`s with `expect`s that describe why this should never fail

Reply:
Bah unfortunately this is `CtOption` but I'll make it better anyway
https://docs.rs/subtle/2.2.2/subtle/struct.CtOption.html

Reply:
wow I can't believe `CtOption` doesn't have an expect call...